src/components/LoginForm/LoginForm.js

```diff
@@ -0,0 +1,293 @@
+
+    data.append('login', loginTextFieldValue);
+    data.append('pass', passwordTextFieldValue);
+
+    return data;
+  };
+
+  makeLoginRequest = async (data) => {
+    try {
+      const response = await axios.post(config.SERVER_URL + 'login', data);
+      this.onSuccessfulResponse(response);
+    } catch (error) {
+      this.onErrorResponse(error);
+    }
+  };
+
+  resetForm = () => {
+    this.setState({
+      isMakingRequest: false,
+      loginTextFieldValue: '',
+      passwordTextFieldValue: '',
+      passwordTextFieldIsDisabled: true,
+    });
+  };
+
+  onSuccessfulResponse = (response) => {
+    const requestNotificationMessage = copyEnum.GENERIC_LOGIN_SUCCESS;
+    this.persistWebTokens(response.data.data);
+    this.showNotification({ requestNotificationMessage }, () => {
+      this.setState({ loginTextFieldIsDisabled: true }, () => setTimeout(this.proceedToMessages, 500));
+    });
+  };
+
+  persistWebTokens = (tokens) => {
+    Object.entries(tokens).forEach(([key, value]) => localStorage.setItem(key, value));
+  };
+
+  proceedToMessages = () => {
+    this.props.showFullscreenLoader();
+    this.props.history.push('/messages');
+  };
+
+  onErrorResponse = (error) => {
+    const errorResponse = (error.response && error.response.data) || {};
+    const requestNotificationMessage = errorResponse.msg || errorsEnum.GENERIC_LOGIN_ERROR;
+
+    this.showNotification(
+      {
+        requestNotificationMessage,
+        requestNotificationType: 'error',
+      },
+      this.resetForm
+    );
+  };
+
+  showNotification = ({ requestNotificationMessage, requestNotificationType = 'success' }, callback = () => null) => {
+    this.setState(
+      {
+        requestNotificationMessage,
+        requestNotificationType,
+        showRequestNotification: true,
+      },
+      () => callback()
+    );
+  };
+
+  closeNotification = () => {
+    this.setState({
+      showRequestNotification: false,
+      requestNotificationMessage: '',
+    });
+  };
+
+  render() {
+    const {
+      isMakingRequest,
+      loginButtonIsDisabled,
+      loginError,
+      loginErrorText,
+      loginTextFieldIsDisabled,
+      loginTextFieldValue,
+      passwordTextFieldValue,
+      passwordTextFieldIsDisabled,
+      requestNotificationMessage,
+      requestNotificationType,
+      showPassword,
+      showRequestNotification,
+    } = this.state;
+
+    return (
+      <Container maxWidth="sm">
+        {' '}
+        //poziom wyżej comtainer z maxWidth md
+        <Card>
+          <CardHeader
+            title="Welcome to Message System"
+            avatar={
+              <Avatar>
+                <Message />
+              </Avatar>
+            }
+          ></CardHeader>
+          <form className="login-form">
+            <div className="login-form-item">
+              <TextField
+                autoComplete="off"
+                id="login"
+                disabled={loginTextFieldIsDisabled}
```

Comment:
So there should be something like that
I guess it's whole condition and `loginTextFieldIsDisabled` mean some external disable comming from props
```suggestion
                disabled={loginTextFieldIsDisabled || isMakingRequest}
```

Reply:
but for now event this will be sufficient
```suggestion
                disabled={isMakingRequest}
```

Reply:
It's all state based  an your suggestions alter logic I assumed. When login success notificiation is displayed I wanted to disable the user from interfering with login form and keep it disabled